v2/inbound_transport_middleware.go

```diff
@@ -68,6 +80,11 @@ func ApplyUnaryInboundTransportMiddleware(handler UnaryTransportHandler, middlew
 // UnaryInboundTransportMiddlewareFunc adapts a function into an InboundMiddleware.
 type UnaryInboundTransportMiddlewareFunc func(context.Context, *Request, *Buffer, UnaryTransportHandler) (*Response, *Buffer, error)
 
+// Name for UnaryInboundTransportMiddlewareFunc
+func (f UnaryInboundTransportMiddlewareFunc) Name() string {
+	return "UnaryInboundTransportMiddlewareFunc"
```

Comment:
Similar to the comment made in another PR, this will be problematic w.r.t multiple middleware registered with the same name.